spec/classes/letsencrypt_scripts_spec.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe 'letsencrypt::scripts' do
+  on_supported_os.each do |os, os_facts|
+    let :facts do
+      os_facts
+    end
```

Comment:
This must be within `context "on #{os}"` or you get very weird results. You essentially only use the last set of facts for all OSes.

Reply:
Moved tests back into `letsencrypt_spec.rb`.